EleksTubeHAX_pio/src/_USER_DEFINES - empty.h

```diff
@@ -33,6 +33,20 @@
 #define BACKLIGHT_DIMMED_INTENSITY  1   // 0..7
 #define TFT_DIMMED_INTENSITY        20  // 0..255
 
+#ifdef HARDWARE_IPSTUBE_CLOCK
+  // comment the next line out to disable hardware dimming with GPIO4 pin (TFT_ENABLE_PIN) for IPSTUBE clock
+  // (in case you have a clock that does not support hardware dimming because of missing Q1 transistor)
+  #define DIM_WITH_ENABLE_PIN_PWM
+  // it is recommended to try this feature only if you are sure that your clock supports hardware dimming
+  // (that is, it is available in native application and really dims the backlight) because it may damage clock
+  // if used with unsupported hardware! this feature only is expected to be supported safely by IPSTUBE clocks by now
+#endif
+
+#ifdef DIM_WITH_ENABLE_PIN_PWM
+  // skip reinitialization if using PWM to control backlight since actual TFTs are never deactivated
+  // disable by commenting the following line if this causes wake up issues from disabled (brightness 0) state
+  #define TFT_SKIP_REINIT
+#endif
```

Comment:
Technically this block should be moved to GLOBAL_DEFINES.h, since we know it's working.

But doesn't matter.

Reply:
Not really, since there can be IPSTUBEs without Q1, and these do not support hardware dimming, and enabling such would disable software one - in such case there will be no dimming at all.

About TFT skip reinit - most likely it should work, but I cannot say that for any hardware possible - maybe better keep it here too for now.

Reply:
I was just wondering if there is a software way to find out if there is a transistor connected to the GPIO. Any ideas?

Reply:
I checked my unfinshed wiring schema and my photos. On the boards, where the Q1 is missing, the GPIO4 is just going to the solder header and ends there...So there can no harm be done, if you use this flag with an non-Q1 version of the clock. 
After checking that, I tried the software on one of this clocks and all is fine. 
You can easily "see" that the transistor is missing, if your night time dimming is not working at all, because the PWM signal goes to nowhere and no software dimming/alpha blending is done anymore...So the displays stays at max brightness all the time...Also the power on/off switch via MQTT/HA is doing nothing...So even if we can't tell apart before, after flashing and trying, you know which version you have...